server.py

```diff
@@ -71,36 +68,22 @@ def execute_user_registration():
 
 
 @app.route("/login", methods=["POST"])
-def check_login_credentials():
+def login():
     """Check user email and password against the db, add user to session."""
 
-    username = request.form.get("username")  # get username from form
-    password = request.form.get("password")  # get password from form
-    user = User.query.filter_by(username=username).first()  # queries database
-
-    if user:  # if the user exists (if that username exists)
-        if user.password == password:  # check if the password is the same
-            session['user'] = user.user_id  # adds user to the session
-
-            flash("You are now logged in.")
-            return redirect("/")
-        else:
-            flash("Incorrect login information. Please try again.")
-            return redirect("/")
-
-    elif not user:
-        flash("Your username does not exist. Please try again or register as a new user.")
-        return redirect("/")
-
-    return render_template("index.html", username=username, password=password,
-                           user=user)
+    user = User.query.filter_by(username=request.form.get('username')).first_or_404()
+    if user.is_valid_password(request.form.get('password')):
```

Comment:
Another example use of the secure password methods.

Also note the example of how to error handle for users that don't exist. It's not necessary here, but I thought it worthwhile to show you other options you have here.